### PR TITLE
feat: Add symfony 6 compatibility

### PR DESCRIPTION
**Pull request by PierreRebeilleau**
| Q             | A
| ------------- | ---
| Branch?       | 2.6 
| Tickets       | https://github.com/api-platform/core/issues/4543, https://github.com/api-platform/core/pull/4581 
| License       | MIT
| Doc PR        | no 

Test are failing because of some type hinting, waiting for a new release of symfony 6 , solved by https://github.com/symfony/symfony/pull/44331

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,12 +18,12 @@
         "fig/link-util": "^1.0",
         "psr/cache": "^1.0 || ^2.0 || ^3.0",
         "psr/container": "^1.0",
-        "symfony/http-foundation": "^4.4 || ^5.1",
-        "symfony/http-kernel": "^4.4 || ^5.1",
-        "symfony/property-access": "^3.4.19 || ^4.4 || ^5.1",
+        "symfony/http-foundation": "^4.4 || ^5.1 || ^6.0",
+        "symfony/http-kernel": "^4.4 || ^5.1 || ^6.0",
+        "symfony/property-access": "^3.4.19 || ^4.4 || ^5.1 || ^6.0",
         "symfony/property-info": "^3.4 || ^4.4 || ^5.2.1",
```
Comment (dunglas) [inline composer.json:24]:
```suggestion
        "symfony/property-info": "^3.4 || ^4.4 || ^5.2.1 || ^6.0",
```

**Review by dunglas — CHANGES_REQUESTED** [review]
Can you also allow the latest versions of the PSR packages please?

**Comment by pietereggink** [comment]
symfony/symfony#44331 is merged, is it possible to execute the quality checks again?

**Comment by rkeet** [comment]
Hi, was looking if I could help out a bit so did some digging through requirements in `composer.json` as I noticed it still has SF 3.4 in there. 

SF 3.4 LTS is now past EOL (since this month). 

Also, SF 3.4 has `"php": "^5.5.9|>=7.0.8"` ([link](https://github.com/symfony/symfony/blob/3.4/composer.json)) for requirements, while ApiPlatform has `">=7.1"` ([link](https://github.com/api-platform/core/blob/3c1f8ece93fc76fb5a3cb91b0f05a161d823deee/composer.json)) (not changed in this PR). Not sure if that would cause issues with compatibility, but maybe a removal of SF 3.4 of supported versions is also an idea for this PR, since you're updating with SF6?

Extra question: is there somewhere I can see/look into/help efforts for a possible v3 and/or minimum PHP 8 / 8.1 requirement?

**Comment by dunglas** [comment]
I would prefer to drop Symfony 3.4/PHP 7.1 in the main branch. But if it's too annoying to keep supporting them, then let's bump these minimal versions.

The main branch contains the 2.7 version, which will be the same code as 3.0 but with a backward compatibility layer for API Platform 2 (that we'll drop before tagging 3.0).

**Review by vincentchalamon — APPROVED** [review]

File: src/GraphQl/Serializer/ItemNormalizer.php [hunk 2]
```diff
@@ -125,6 +129,10 @@ protected function setAttributeValue($object, $attribute, $value, $format = null
             $attribute = 'id';
         }
 
+        if ('{}' === $value) {
```
Comment (dunglas) [inline src/GraphQl/Serializer/ItemNormalizer.php:132]: That's weird. Why do we need this?

File: tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php [hunk 3]
```diff
@@ -808,49 +808,18 @@ public function testDisabledPaginationViaDefaults()
 
     public function testItLoadsMetadataConfigFileInAlphabeticalOrder()
     {
-        $yamlExtractorDefinition = $this->prophesize(Definition::class);
```
Comment (dunglas) [inline tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php:811]: Why is this test removed?

**Review by dunglas — CHANGES_REQUESTED** [review]

File: src/GraphQl/Serializer/ItemNormalizer.php (hunk 2 above)
Comment (PierreRebeilleau) [reply src/GraphQl/Serializer/ItemNormalizer.php:132]: When we dump `dump($context['args']['input']);` in  the DeserializeStage we got this : 
```
array:4 [
  "id" => "/related_dummies/2"
  "symfony" => "laravel"
  "thirdLevel" => "/third_levels/1"
  "embeddedDummy" => "{}"
]
```

Comment (dunglas) [reply src/GraphQl/Serializer/ItemNormalizer.php:132]: This looks like a bug in pho-GraphQL. Could you report it upstream and explain why we need this temporary fix in a comment please?

Comment (alanpoulain) [reply src/GraphQl/Serializer/ItemNormalizer.php:132]: Input is user-defined, there shouldn't be a fix in the serializer. Maybe the test is bad. Which test is failing?

Comment (PierreRebeilleau) [reply src/GraphQl/Serializer/ItemNormalizer.php]: https://github.com/api-platform/core/runs/4595101310?check_suite_focus=true
Seems to be this one

Comment (alanpoulain) [reply src/GraphQl/Serializer/ItemNormalizer.php]: This is strange, we need to dig it, it's probably not the right fix.

**Comment by alanpoulain** [comment]
Thank you everyone.

**Comment by dunglas** [comment]
You rock!

**Pull request merged**